test-data/unit/deps-types.test

```diff
@@ -152,6 +152,140 @@ def h() -> None:
 <m.ff> -> m, m.h
 <m.g> -> m.f
 
+[case testMetaclassAttributes]
+from mod import C
+from typing import Type
+def f(arg: Type[C]) -> None:
+    arg.x
+[file mod.py]
+class M(type):
+    x: int
+class C(metaclass=M):
+    pass
+[out]
+<mod.C.x> -> m.f
+<mod.C> -> <m.f>, m, m.f
+<mod.M.x> -> m.f
+
+[case testMetaclassAttributesDirect]
+from mod import C
+def f() -> None:
+    C.x
+[file mod.py]
+class M(type):
+    x: int
+class C(metaclass=M):
+    pass
+[out]
+<mod.C.__init__> -> m.f
+<mod.C.x> -> m.f
+<mod.C> -> m, m.f
+<mod.M.x> -> m.f
+
+[case testMetaclassOperators]
+from mod import C
+from typing import Type
+def f(arg: Type[C]) -> None:
+    arg + arg
+[file mod.py]
+class M(type):
+    def __add__(self, other: M) -> M:
+        pass
+class C(metaclass=M):
+    pass
+[out]
+<mod.C> -> <m.f>, m, m.f
+<mod.M.__add__> -> m.f
+<mod.M.__radd__> -> m.f
+
+[case testMetaclassOperatorsDirect]
+from mod import C
+def f() -> None:
+    C + C
+[file mod.py]
+class M(type):
+    def __add__(self, other: M) -> M:
+        pass
+class C(metaclass=M):
+    pass
+[out]
+<mod.C.__init__> -> m.f
+<mod.C> -> m, m.f
+<mod.M.__add__> -> m.f
+<mod.M.__radd__> -> m.f
+
+[case testMetaclassAttributes_python2]
+# flags: --py2
+from mod import C
+from typing import Type
+def f(arg):
+    # type: (Type[C]) -> None
+    arg.x
+[file mod.py]
+class M(type):
+    x = None  # type: int
+class C:
+    __metaclass__ = M
+[out]
+<mod.C.x> -> m.f
+<mod.C> -> <m.f>, m, m.f
+<mod.M.x> -> m.f
+
+[case testMetaclassAttributesDirect_python2]
+# flags: --py2
+from mod import C
+def f():
+    # type: () -> None
+    C.x
+[file mod.py]
+class M(type):
+    x = None  # type: int
+class C:
+    __metaclass__ = M
+[out]
+<mod.C.__init__> -> m.f
+<mod.C.x> -> m.f
+<mod.C> -> m, m.f
+<mod.M.x> -> m.f
+
+[case testMetaclassOperators_python2]
+# flags: --py2
+from mod import C
+from typing import Type
+def f(arg):
+    # type: (Type[C]) -> None
+    arg + arg
+[file mod.py]
+class M(type):
+    def __add__(self, other):
+        # type: (M) -> M
+        pass
+class C:
+    __metaclass__ = M
+[out]
+<mod.C> -> <m.f>, m, m.f
+<mod.M.__add__> -> m.f
+<mod.M.__radd__> -> m.f
+
+[case testMetaclassOperatorsDirect_python2]
+# flags: --py2
+from mod import C
+def f():
+    # type: () -> None
+    C + C
+[file mod.py]
+class M(type):
+    def __add__(self, other):
+        # type: (M) -> M
+        pass
+class C:
+    __metaclass__ = M
+[out]
+<mod.C.__init__> -> m.f
+<mod.C> -> m, m.f
+<mod.M.__add__> -> m.f
+<mod.M.__radd__> -> m.f
```

Comment:
Also test dependency from metaclass to the class that has the metaclass. Could be something like `<M> -> <C>`.

Reply:
I think I actually missing this dependency. Namely, there are two metaclasses, declared and actual (calculated). I think we don't need deps on the latter (also it appears as a part of class snapshot), but I think we need a dep on declared metaclass.